reference/enterprise-api-ref/federated-reporting-api.markdown

```diff
@@ -0,0 +1,226 @@
+---
+layout: default
+title: Federated reporting configuration API
+published: true
+tags: [reference, enterprise, API, reporting]
+---
+
+Federated reporting API allows to configure federated reporting process.
+
+# Remote hubs
+
+## Remote hubs list
+
+**URI:** https://hub.cfengine.com/api/fr/remote-hub
+
+**Method:** GET
+
+**Example response:**
+
+```
+HTTP 200 OK
+{
+    "id-1": {
+        "id": 1,
+        "hostkey": "SHA=2d67a6840878de...",
+        "api_url": "https://172.28.128.5",
+        "ui_name": "ubuntu-xenial",
+        "role": "feeder",
+        "target_state": "on",
+        "transport": {
+            "mode": "pull_over_rsync",
+            "ssh_user": "cfdrop",
+            "ssh_host": "172.28.128.5",
+            "ssh_pubkey": "",
+            "ssh_fingerprint": ""
+        }
+    },
+    "id-2": {
+        "id": 2,
+        "hostkey": "SHA=wefweg34tgfds...",
+        "api_url": "https://172.28.128.6",
+        "ui_name": "ubuntu-beaver",
+        "role": "feeder",
+        "target_state": "on",
+        "transport": {
+            "mode": "pull_over_rsync",
+            "ssh_user": "cfdrop",
+            "ssh_host": "superhub",
+            "ssh_pubkey": "pubkey cfdrop",
+            "ssh_fingerprint": null
+        }
+    }
+}
+```
+
+## Get remote hub
+
+**URI:** https://hub.cfengine.com/api/fr/remote-hub/:remote_hub_id
+
+**Method:** GET
+
+**Parameters:**
+
+* **remote_hub_id** *(number)*
+    Remote hub id
+
+**Example response:**
+
+```
+HTTP 200 OK
+{
+    "id": 1,
+    "hostkey": "SHA=2d67a6840878de098abbef1172f103a6febbfb5d00b8ace31ca3d46a9d22930d",
+    "api_url": "https://172.28.128.5",
+    "ui_name": "ubuntu-xenial",
+    "role": "feeder",
+    "target_state": "on",
+    "transport": {
+        "mode": "pull_over_rsync",
+        "ssh_user": "cfdrop",
+        "ssh_host": "172.28.128.5",
+        "ssh_pubkey": "",
+        "ssh_fingerprint": ""
+    }
+}
+```
+
+## Add remote hub
+
+**URI:** https://hub.cfengine.com/api/fr/remote-hub
+
+**Method:** POST
+
+**Parameters:**
+
+* **ui_name** *(string)*
+    Remote hub name
+* **hostkey** *(string)*
+    Remote hub hostkey
+* **role** *(string)*
+    Remote hub role. Allowed values: `feeder`, `superhub`
+* **target_state** *(string)*
+    Target state of remote hub. Allowed values: `on`, `paused`
+* **transport** *(json)*
+    Transport data. Emp `{ "mode": "pull_over_rsync", "ssh_user": "cfdrop", "ssh_host": "172.28.128.5", "ssh_pubkey": "", "ssh_fingerprint": ""}`
+
+**Example response:**
+
+```
+HTTP 201 CREATED
+```
```

Comment:
I think there should be a note here that the config file is not changed until the `/federation-config` API request is done. And the same applies to all the other requests that change the configuration.

Reply:
@vpodzime done, please review